kafka/conn.py

```diff
@@ -1040,6 +1040,13 @@ def send_pending_requests_v2(self):
                     total_bytes = self._send_bytes(self._send_buffer)
                     self._send_buffer = self._send_buffer[total_bytes:]
 
+                # If all data was sent, we need to get the new data from the protocol now, otherwise
+                # this function would return True, indicating that there are no more pending
+                # requests. This could cause the calling thread to wait indefinitely as it won't
+                # know that there is still buffered data to send.
+                if not self._send_buffer:
+                    self._send_buffer = self._protocol.send_bytes()
```

Comment:
I think I understand the idea here.

But maybe a cleaner fix would be to change:

```python
return len(self._send_buffer) == 0
```

To 

```python
return not self._protocol.has_bytes_to_send()
```

And add in `KafkaProtocol`

```python
    def has_bytes_to_send(self) -> bool:
        """Check if there are any bytes pending to send on the network"""
        return len(self.bytes_to_send) > 0
```

Reply:
This looks like a quite bad bug, ideas how it seems to still exist in `kafka-python`? Does it only happen when `                    total_bytes = self._send_bytes(self._send_buffer)` doesn't send the full buffer?